src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -232,6 +234,44 @@ BinaryCompactObject::BinaryCompactObject(
   } catch (const std::exception& error) {
     PARSE_ERROR(context, "Invalid 'InitialGridPoints': " << error.what());
   }
+
+  // Compute the inner radius of the outer spherical shell. The computation
+  // makes use of the refinement, so this can't be done earlier.
+  if (radius_enveloping_sphere.has_value()) {
+    radius_enveloping_sphere_ = radius_enveloping_sphere.value();
+    if (radius_enveloping_sphere_ <= radius_enveloping_cube_ or
+        radius_enveloping_sphere_ >= outer_radius_) {
+      PARSE_ERROR(
+          context,
+          "The 'OuterShell.InnerRadius' must be within 'EnvelopingCube.Radius' "
+          "(" << radius_enveloping_cube_
+              << ") and 'OuterShell.OuterRadius' (" << outer_radius_
+              << "), but is: " << radius_enveloping_sphere_
+              << ". Set it to 'Auto' so a reasonable value is chosen "
+                 "automatically.");
+    }
+  } else {
+    // Adjust the outer boundary of the cubed sphere to conform to the spacing
+    // of the spherical shells after refinement, so the cubed sphere is the same
+    // size as the first radial division of the spherical shell (for linear
+    // mapping) or smaller by the same factor as adjacent radial divisions in
+    // the spherical shell (for logarithmic or inverse mapping).
+    const size_t addition_to_outer_layer_radial_refinement_level =
+        initial_refinement_[44][2] - initial_refinement_[34][2];
+    const double radial_divisions_in_outer_layers =
+        pow(2, addition_to_outer_layer_radial_refinement_level) + 1;
+    const double singularity_pos = 0.;
+    radius_enveloping_sphere_ =
+        domain::CoordinateMaps::Interval{// Source interval
+                                         0., radial_divisions_in_outer_layers,
+                                         // Target interval
+                                         radius_enveloping_cube_, outer_radius_,
+                                         // Distribution in target interval
+                                         radial_distribution_outer_shell_,
+                                         singularity_pos}(
+            // Inner radius in source interval
```

Comment:
is `1.` the `Inner radius in source interval`, or is `std::array<double, 1>{{1.}})[0]`? (I assume `std::array<double, 1>{{1.}})[0]` is the inner radius in the target interval, but hopefully the comment can be made more clear

Reply:
See above (and fixup)